pkg/apiserver/resthandler.go

```diff
@@ -519,3 +534,22 @@ func setListSelfLink(obj runtime.Object, req *restful.Request, namer ScopeNamer)
 	return runtime.SetList(obj, items)
 
 }
+
+func getPatchedJS(contentType string, originalJS, patchJS []byte, obj runtime.Object) ([]byte, error) {
```

Comment:
I agree with the abstraction but it makes deciding where to put PatchType a little more difficult. If PatchType stays in the client, it probably is weird for resthandler to reference the client package to do its work. But if PatchType moves to the server, the client has to reference internal server details to construct a request. Hmm...